### PR TITLE
A Guide to Run your React Native Apps with Typescript

### LINKED ISSUE #5851: A Guide to Run your React Native Apps with Typescript
#### Proposed title of the article

An Absolute Beginner Guide to Run your React Native Apps with Typescript

#### Proposed article introduction

React Native runs of JavaScript by default. However, to leverage the advantages that Typescript (a superset of JavaScript) offers, it is possible to set up your whole React Native apps to run wholes on Typescript. This guide aims to thoroughly introduce the reader to running React Native apps with Typescript. We will learn the advantages of running React Native apps with Typescript and how to set them up with React Native. In the end, we will create a very handy blog application that uses Typescript to run the React Native application.

#### Key takeaways 

- What is Typescript
- Why run apps with Typescript
- Getting Started with Typescript
- What is React Native
- Why use React to run Native apps
- Initializing your First React Native project with Typescript
- Set up a Typescript Blog app using React Native
- Create React Native blog app components
- Add posts to the application
- Fetch all added posts
- Fetch a single added post

#### Article quality

This guide will be a thoughtful introduction to Typescript and React Native. For better understanding, the reader will also be able to create a whole application that uses React Native to run the Typescript code.

### PR DESCRIPTION
**Pull request by kimkimani**
#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #5851

### CODE REVIEW

**Comment by WanjaMIKE** [comment]
In your TypeScript objects, please use original content rather than the current ones. They are showing a high level of plagiarism. Thanks

[New Scan 5_49 PM Copyleaks Report.pdf](https://github.com/section-engineering-education/engineering-education/files/7864929/New.Scan.5_49.PM.Copyleaks.Report.pdf)

**Comment by Jethro-magaji** [comment]
> In your TypeScript objects @kimkimani , please use original content rather than the current ones. They are showing a high level of plagiarism. Thanks
> 
> [New Scan 5_49 PM Copyleaks Report.pdf](https://github.com/section-engineering-education/engineering-education/files/7864929/New.Scan.5_49.PM.Copyleaks.Report.pdf)

@kimkimani Make the edits so that I can carry on with the review process.

**Comment by Jethro-magaji** [comment]
Hi @kimkimani 
Well done for coming up with such an amazing article but however, there're a few grammatical errors. Please use 2nd person singular "You" instead of "We", also I highly suggest you use tools like [Grammarly](https://grammarly.com/) and [ProWritingAid ](https://prowritingaid.com/Free) to edit your work.
Make these changes and I will continue the review process.
Thanks.

File: content/articles/guide-to-run-react-native-app-with-typescript/index.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-React Native runs of JavaScript by default. However, to leverage Typescript's advantages (a superset of JavaScript) offers, it is possible to set up your whole React Native apps to run entirely on Typescript. This guide aims to briefly introduce you to running React Native apps with Typescript. We will learn the advantages of running React Native apps with Typescript and how to set them up with React Native. In the end, we will create a very handy blog application that uses Typescript to run the React Native application.
+React Native runs of JavaScript by default. However, to leverage Typescript's advantages (a superset of JavaScript) offers, it is possible to set up your whole React Native apps to run entirely on Typescript. This guide aims to briefly introduce you to running React Native apps with Typescript. You will learn the advantages of running React Native apps with Typescript and how to set them up with React Native. In the end, you will create a very handy blog application that uses Typescript to run the React Native application.
```
Comment (Jethro-magaji) [inline content/articles/guide-to-run-react-native-app-with-typescript/index.md:1]: React Native runs of JavaScript by default; Do you mean "runs on" or "runs of"? I think it should be "runs on".

File: content/articles/guide-to-run-react-native-app-with-typescript/index.md [hunk 2]
```diff
@@ -1,4 +1,4 @@
...
 
 ### Requirements
```
Comment (Jethro-magaji) [inline content/articles/guide-to-run-react-native-app-with-typescript/index.md:3]: Please rename it to "Pre-requisites".

File: content/articles/guide-to-run-react-native-app-with-typescript/index.md [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+  },
+});
+```
+
+Your posts should now be shown on your screen:
+
+![posts-screen](/engineering-education/guide-to-run-react-native-app-with-typescript/posts-screen.jpg)
+
+### Conclusion
+This guide walked you through a reflective introduction to Typescript and React Native. For better understanding, we have created a whole application that uses React Native to run the Typescript code.
```
Comment (Jethro-magaji) [inline content/articles/guide-to-run-react-native-app-with-typescript/index.md:349]: Include any next or final steps summarizing the tasks the audience just completed with possible next steps.
Also, include a link to the Github repo or live site, if you can.

**Review by Jethro-magaji — CHANGES_REQUESTED** [review]
Hi @kimkimani 
Well done for the edits. However, please make these minor changes.
Thanks.

**Comment by kimkimani** [comment]
@Jethro-magaji Check this out

**Review by Jethro-magaji — APPROVED** [review]
Hi! @kimkimani 
Well done for such an amazing, please take note of the reviews I made and effect them in your next article to minimize revisions next time. 
Thanks.

**Comment by WanjaMIKE** [comment]
Merged #6293 into master

**Comment by WanjaMIKE** [comment]
@Jethro-magaji ensure that you put the correct author name in the front matter.

**Comment by Jethro-magaji** [comment]
> @Jethro-magaji ensure that you put the correct author name in the front matter.

Noted, thanks

**Pull request merged**